studio/data/database/schemas-query.ts

```diff
@@ -1,60 +1,50 @@
-import { useCallback } from 'react'
-import { UseQueryOptions } from '@tanstack/react-query'
-import { ExecuteSqlData, useExecuteSqlPrefetch, useExecuteSqlQuery } from '../sql/execute-sql-query'
-
-export type Schema = {
-  name: string
-}
-
-export const getSchemasQuery = () => {
-  const sql = /* SQL */ `
-  SELECT nspname as name
-  FROM pg_namespace
-  WHERE
-    nspname not in ('information_schema', 'pg_catalog', 'pg_toast')
-    AND nspname not like 'pg_temp_%'
-    AND nspname not like 'pg_toast_temp_%'
-    AND has_schema_privilege(oid, 'CREATE, USAGE')
-  ORDER BY nspname;
-`.trim()
-
-  return sql
-}
+import { PostgresSchema } from '@supabase/postgres-meta'
+import { useQuery, UseQueryOptions } from '@tanstack/react-query'
+import { get } from 'lib/common/fetch'
+import { API_URL } from 'lib/constants'
+import { databaseKeys } from './keys'
 
 export type SchemasVariables = {
   projectRef?: string
   connectionString?: string
 }
 
-export type SchemasData = { result: Schema[] }
-export type SchemasError = unknown
+export type Schema = PostgresSchema
+
+export type SchemasResponse = Schema[] | { error?: any }
 
-export const useSchemasQuery = <TData extends SchemasData = SchemasData>(
+export async function getSchemas(
   { projectRef, connectionString }: SchemasVariables,
-  options: UseQueryOptions<ExecuteSqlData, SchemasError, TData> = {}
-) => {
-  return useExecuteSqlQuery(
-    {
-      projectRef,
-      connectionString,
-      sql: getSchemasQuery(),
-      queryKey: ['schemas'],
-    },
-    options
-  )
+  signal?: AbortSignal
+) {
+  if (!projectRef) {
+    throw new Error('projectRef is required')
+  }
+
+  let headers = new Headers()
+  if (connectionString) headers.set('x-connection-encrypted', connectionString)
+
+  const response = (await get(`${API_URL}/pg-meta/${projectRef}/schemas`, {
```

Comment:
just to sanity check that this should be fine? cause the old schemas query we had were filtering pg_temp and pg_toast_temp

Reply:
Yep this endpoint doesn't seem to return those schemas